lib/angular2/shared/actions/model.ejs

```diff
@@ -0,0 +1,258 @@
+/* tslint:disable */
+import { Action } from '@ngrx/store';
+import { type } from '../util';
+import { BaseLoopbackActionTypesFactory, BaseLoopbackActionsFactory } from './base';
+import { LoopBackFilter, SDKToken, <%- modelName %> } from '../models';
+
+export const <%- modelName %>ActionTypes =
+Object.assign(BaseLoopbackActionTypesFactory('<%- modelName %>'), {
+<% model.methods.forEach(function(action) {
+ 
+  var methodName = action.name.split('.').join('$').replace('prototype$', '').replace(/::/g, '__');
+  var httpVerb = action.getEndpoints ? action.getEndpoints()[0].verb : action.getHttpMethod();
+  var fullPath = action.getEndpoints ? action.getEndpoints()[0].fullPath : action.getFullPath();
+
+  // all of these methods are extended from base service
+  if (methodName.match(/(^create$|^createMany$|^find$|^replaceOrCreate$|^replaceById$|^upsert$|^upsertWithWhere$|^exists$|^findOne$|^findById$|^deleteById$|^updateAttributes$|^patchOrCreate$|^patchAttributes$|^updateAll$|^count$|^createChangeStream$)/)) { return; }
```

Comment:
I see this file was copied from the services one, and I see a lot of repeated code. Maintaining both will be difficult I will recommend to decouple all the repeated lines in this file and the services one and then create a helper that provides the same base code.